src/main/scala/temple/generate/server/go/service/GoServiceDAOGenerator.scala

```diff
@@ -51,10 +51,14 @@ object GoServiceDAOGenerator {
   private def generateDatastoreInterfaceFunction(
     serviceName: String,
     operation: CRUD,
-    enumByCreatedBy: Boolean,
+    createdByAttribute: CreatedByAttribute,
   ): String = {
     val functionName = generateDAOFunctionName(operation, serviceName)
-    val functionArgs = if (enumByCreatedBy || operation != CRUD.ReadAll) s"input ${functionName}Input" else ""
+    val enumeratingByCreator = createdByAttribute match {
```

Comment:
Is there a reason why this isn’t `createdByAttribute.isInstanceOf[CreatedByAttribute.EnumerateByCreator]`?

Reply:
Discussed offline, essentially isInstanceOf uses reflection under the hood, and not pattern matching for some reason